packaging/olm-manifests/generate-manifests.sh

```diff
@@ -0,0 +1,152 @@
+  # Update deployment
+  yq ea -i 'select(fi==0).spec.install.spec.deployments[0].spec = select(fi==1).spec | select(fi==0)' ${CSV_FILE} ${CRD_DIR}/060-Deployment-strimzi-cluster-operator.yaml
+  yq ea -i ".spec.install.spec.deployments[0].name = \"${BUNDLE_NAME}\"" ${CSV_FILE}
+  # Update namespaced cluster roles
+  yq ea -i 'select(fi==0).spec.install.spec.permissions[0].rules = select(fi==1).rules | select(fi==0)' ${CSV_FILE} ${CRD_DIR}/020-ClusterRole-strimzi-cluster-operator-role.yaml
+  # Update global cluster roles
+  yq ea -i 'select(fi==0).spec.install.spec.clusterPermissions[0].rules = select(fi==1).rules | select(fi==0)' ${CSV_FILE} ${CRD_DIR}/021-ClusterRole-strimzi-cluster-operator-role.yaml
+  # Update creation timestamp
+  yq ea -i ".metadata.annotations.createdAt = \"$(date +'%Y-%m-%d %H:%M:%S')\"" ${CSV_FILE}
+  
+  generate_related_images
+  generate_image_digests
+}
+
+# The `spec.relatedImages` field is a specific to OpenShift and not
+# in the operator-sdk spec, so we have to generate it ourselves.
+generate_related_images() {
+  yq ea -i '.spec.relatedImages = null' ${CSV_FILE}
+
+  ENV="*-operator *BRIDGE_IMAGE *JMXTRANS_IMAGE *EXECUTOR_IMAGE"
+  for env in $ENV; 
+  do
+    image=$(yq ea ".. | select(has(\"name\")).env[] | (select (.name == \"$env\")).value" ${CSV_FILE})
+    if [ -z "$image" ]; then
+      # Find operator image
+      image=$(yq ea ".. | select(has(\"image\")) | (select (.name == \"$env\")).image" ${CSV_FILE} | head -n 1) 
+    fi
+    name="strimzi-$(echo $image | cut -d':' -f1 | cut -d'@' -f1 | rev | cut -d'/' -f1 | rev)";
+    yq ea -i ".spec.relatedImages += [{\"name\": \"$name\", \"image\": \"$image\"}]" ${CSV_FILE};
+  done
+
+  # Add Kafka images to relatedImages section.
+  KAFKA_IMAGE_VALUES=$(yq eval '.. | select(has("name")).env[] | (select (.name == "STRIMZI_KAFKA_IMAGES")).value' ${CSV_FILE})
+  for val in $KAFKA_IMAGE_VALUES;
+  do
+    name="strimzi-kafka-$(echo "$val" | cut -d'=' -f1 | tr -d '.')";
+    image=$(echo "${val}" | cut -d'=' -f2);
+    yq ea -i ".spec.relatedImages += [{\"name\": \"$name\", \"image\": \"$image\"}]" ${CSV_FILE};
+  done
+}
+
+# Iterates through images listed in newly generated relatedImages section
+# and replaces image tags with image digests
+generate_image_digests() {
+
+  IMAGES=$(yq eval '.spec.relatedImages[].image' "${CSV_FILE}")
+  for image_tag in $IMAGES;
+  do
+    tag=$(echo $image_tag | cut -d':' -f2);
+    image=$(echo $image_tag | cut -d'@' -f1 | cut -d':' -f1);
+    
+    registry=$(echo $image | cut -d'/' -f1);
+    org=$(echo $image | rev | cut -d'/' -f2 | rev);
+    repo=$(echo $image | rev | cut -d'/' -f1 | rev);
+    
+    echo "Get digest from remote registry for: $image_tag"; 
+    digest="$(curl -s -H "Accept: application/vnd.docker.distribution.manifest.v2+json" https://${registry}/v2/${org}/${repo}/manifests/${tag} | sha256sum | head -c 64)";
```

Comment:
It will retrieve the image specified in the yaml files, more specifically the `packaging/install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml` file. 

The script assumes that the generated deployment yaml has the latest images

Reply:
I'm not completely sure this is the case with:
* The images often being pushed into different repositories
* The images being pushed only at the end of the build

Reply:
We could restrict the generator script to only be used as part of the Strimzi release workflow after the deployment yaml has been updated to point to the officially supported images of the release. Would this solve the problem?

Of course, the logic can be changed to retrieve the image path information from the parent `Makefile.docker` file but it would be a little more complicated. Would that be preferable?

Reply:
Well, I'm not sure what would solve the problem TBH. It depends how and what for is it supposed to be used. Maybe it should be moved to `tools` to make it clear it is not used anywhere during the build / release process but it is just a tool for a manual use?

Reply:
I have moved the generator scripts to the tools/olm-manifest directory for manual execution for now

Reply:
@kyguy could we not inspect the various `$DOCKER_xxxxxxxx` env vars? You may have to check where in the build chain the defaults are set though?

Reply:
@tomncooper I am not sure I understand, by `$DOCKER_xxxxxxxx` env vars do you mean the env vars listed in the `packaging/install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml` file?

Reply:
No, in the docker build script we use `DOCKER_REGISTRY`, `DOCKER_ORG` and `DOCKER_TAG` env vars to control how images are named, tagged and what repo they are pushed too. If these are set for a custom build then you can construct the container image address from them and pull the image that was just built.

Reply:
Thanks for the clarification Tom! I have just added a Makefile in the `tools/olm-bundle` directory which will pulls in the env vars from the parent `Makefile.docker` file. Now we can execute the generator by running:
```
make BUNDLE_VERSION=0.23.0
```
and it will use the env vars defined in the Makefile.docker file: `DOCKER_REGISTRY`, `DOCKER_ORG`, and `DOCKER_TAG` to populate the image fields in the CSV file.